contracts/src/Zappers/Modules/Exchanges/HybridCurveUniV3ExchangeHelpers.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.18;
+
+import "../../../Interfaces/IWETH.sol";
+// Curve
+import "./Curve/ICurveStableswapNGPool.sol";
+// UniV3
+import "./UniswapV3/IQuoterV2.sol";
+
+import "../../Interfaces/IExchangeHelpers.sol";
+
+contract HybridCurveUniV3ExchangeHelpers is IExchangeHelpers {
+    uint256 private constant DECIMAL_PRECISION = 1e18;
+
+    //HybridCurveUniV3Exchange public immutable exchange;
+
+    IERC20 public immutable USDC;
+    IWETH public immutable WETH;
+
+    // Curve
+    ICurveStableswapNGPool public immutable curvePool;
+    uint128 public immutable USDC_INDEX;
+    uint128 public immutable BOLD_TOKEN_INDEX;
+
+    // Uniswap
+    uint24 public immutable feeUsdcWeth;
+    uint24 public immutable feeWethColl;
+    IQuoterV2 public immutable uniV3Quoter;
+
+    /*
+    constructor(HybridCurveUniV3Exchange _exchange) {
+        exchange = _exchange;
+
+        USDC = _exchange.USDC();
+        WETH = _exchange.WETH();
+
+        curvePool = _exchange.curvePool();
+        USDC_INDEX = _exchange.USDC_INDEX();
+        BOLD_TOKEN_INDEX = _exchange.BOLD_INDEX();
+
+        // Uniswap
+        feeUsdcWeth = _exchange.feeUsdcWeth();
+        feeWethColl = _exchange.feeWethColl();
+        uniV3Quoter = _exchange.uniV3Quoter();
+    }
+    */
+
+    constructor(
+        IERC20 _usdc,
+        IWETH _weth,
+        // Curve
+        ICurveStableswapNGPool _curvePool,
+        uint128 _usdcIndex,
+        uint128 _boldIndex,
+        // UniV3
+        uint24 _feeUsdcWeth,
+        uint24 _feeWethColl,
+        IQuoterV2 _uniV3Quoter
+    ) {
+        USDC = _usdc;
+        WETH = _weth;
+
+        // Curve
+        curvePool = _curvePool;
+        USDC_INDEX = _usdcIndex;
+        BOLD_TOKEN_INDEX = _boldIndex;
+
+        // Uniswap
+        feeUsdcWeth = _feeUsdcWeth;
+        feeWethColl = _feeWethColl;
+        uniV3Quoter = _uniV3Quoter;
+    }
+
+    function getCollFromBold(uint256 _boldAmount, IERC20 _collToken, uint256 _desiredCollAmount) external /* view */ returns (uint256, uint256) {
+        // BOLD -> USDC
+        uint256 curveUsdcAmount = curvePool.get_dy(int128(BOLD_TOKEN_INDEX), int128(USDC_INDEX), _boldAmount);
+
+        // USDC -> Coll
+        bytes memory path;
+        if (address(WETH) == address(_collToken)) {
+            path = abi.encodePacked(USDC, feeUsdcWeth, WETH);
+        } else {
+            path = abi.encodePacked(USDC, feeUsdcWeth, WETH, feeWethColl, _collToken);
+        }
+
+        (uint256 collAmount,,,) = uniV3Quoter.quoteExactInput(path, curveUsdcAmount);
+
+        uint256 slippage = collAmount > _desiredCollAmount ? 0: DECIMAL_PRECISION - collAmount * DECIMAL_PRECISION / _desiredCollAmount;
```

Comment:
I question the usefulness of `slippage` being calculated in the helper. If this is about calculating the price impact of the trade, the frontend would first have to call `getCollFromBold()` passing some small amount of BOLD as input (e.g. $1 BOLD) to get the current BOLD:COLL "spot price", from which `_desiredCollAmount` can be calculated. Then it'd have to call `getCollFromBold()` a second time, now passing a meaningful `_desiredCollAmount`. However, it's more efficient to call `getCollFromBold()` twice in parallel, as it can be performed in a single multicall, then performing the very simple calculation on the client side.

Is there some other use of `_desiredCollAmount` and `slippage` that I'm missing?

Reply:
Having read the tests, I guess the idea is to use the oracle price of the collateral as a basis for calculating `_desiredCollAmount`, but I have 2 issues with that:
- The oracle price is in USD, so we're ignoring the BOLD:USD price.
- Calling it slippage is IMO misleading, as that normally refers to the difference between the exchange rate as predicted and the exchange rate at which the trade is actually executed. What we're calculating here is quite far removed from that.

Reply:
Okay, now I think it's definitely useful to calculate this when you're leveraging, as any deviation from the oracle price on the downside will eat into your leverage, I'm just not sure:
- if slippage is the right term
- if it needs to be included in the helper (the frontend already has to perform half of the calculation to make it work, by fetching the oracle price and calculating `_desiredCollAmount` accordingly)

Reply:
> I guess the idea is to use the oracle price of the collateral as a basis for calculating `_desiredCollAmount`

That’s the initial approach, but this function is agnostic to that. It may be even be a value provided by the user. So, the function how the right price is determined.

I agree that the term slippage is not the best one, I’ll change it. It can also be removed, yes, but I just thought it was more convenient, even for manual calls. Do you think it hurts to have it there?

Reply:
As long as it's optional (i.e. we can safely pass 0) I don't think it hurts. I just wouldn't use it if I was writing a frontend, as it introduces a point of serialization (you have to wait for the result of whatever query you want to use to determine the right price before you can make the call) which would make the frontend less responsive.